File: TD/TD7/Code/7.cpp
/**
 * Remplir un tableau avec les coefficients du triangle de Pascal
 */
#include "sunlib.hpp" // Contient la fonction combin()
#include <iostream>

const int MAX = 100;

/* Procédure */

void trianglePascal(int tab[MAX][MAX], int n)
{
  for (int i = 0; i < n; ++i)
    for (int j = 0; j < i + 1; ++j)
      tab[i][j] = combin(i, j);
}

/* Utilisation */

int main()
{
  int tab[MAX][MAX];
  
  trianglePascal(tab, 6);
  
  // Affichage de ce tableau
  for (int i = 0; i < 6; ++i) {
    for (int j = 0; j < i + 1; ++j)
      std::cout << tab[i][j] << "\t" << std::flush;
    std::cout << std::endl;
  }
  
  return 0;
}

File: TD/TD7/Code/Python/7.py
#!/usr/bin/python
# Remplir un tableau avec les coefficients du triangle de Pascal
from math import factorial as fact


def combin(n, p):
    res = int(fact(n) / (fact(p) * fact(n - p)))
    return res


def triangle_pascal(tab):
    for i in range(len(tab)):
        for j in range(i + 1):
            tab[i][j] = combin(i, j)


tab = [[[] for i in range(6)] for j in range(6)]

triangle_pascal(tab)

# Affichage de tableau
for k in range(len(tab)):
    for m in range(k + 1):
        print(tab[k][m], '', sep='\t', end='')
    print('')
